### ISSUE TITLE
Initialization of container $containerName failed

### ISSUE THREAD

**Issue by rdebath** [microsoft/navcontainerhelper#2663]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
Hi, Any ideas from the following ?

```
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.12\ContainerHandling\Wait-NavContainerReady.ps1:42
char:17
+ ...             throw "Initialization of container $containerName failed" ...
```
Occurs for both `master` and `NextMajor`.
Occurs for both multitenant true and false
Occurs for Dev License and Cronus license (default)
Before and after `Update-Module bccontainerhelper`

Last master I got was was 21.0.43576-21.0.43463 on 2022-08-10
Works for BCArtifactUrl: https://bcartifacts.azureedge.net/sandbox/20.4.44313.45693/gb

Container appears to continue running, but `docker logs` shows nothing more.

```
PS C:\tvt> docker ps -a
CONTAINER ID   IMAGE                                               COMMAND                  CREATED         STATUS                   PORTS                                                                                                      NAMES
42265de9d658   mcr.microsoft.com/businesscentral:10.0.14393.4104   "powershell -Command…"   5 minutes ago   Up 5 minutes (healthy)   0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp, 1433/tcp, 7083/tcp, 0.0.0.0:7045-7049->7045-7049/tcp, 8080/tcp   bcserver
PS C:\tvt>
```

With `dumpEventLog` turned on I only get extra repeated:
```
TimeGenerated : 08/31/2022 18:07:51
EventSource: MicrosoftDynamicsNavServer$BC
EntryType : Warning
Message :
Server instance: BC
Category: Metadata
ClientSessionId: 00000000-0000-0000-0000-000000000000
ClientActivityId: ad27f896-af56-46cc-9d66-eb1a8d67ed8a
ServerSessionUniqueId: 00000000-0000-0000-0000-000000000000
ServerActivityId: a9d9e022-90a3-4017-bb34-c0173bb624ee
EventTime: 08/31/2022 17:07:51
Message The object name 'VAT Group Authentication Type ' can only appear once for the type 'Enum'. The same object exist
s in both extension ('c50a4bf0-db51-4ad2-88d5-fe2287da0eb8' 'VAT Group Management', Publisher: Microsoft, Package ID: 17
7203f4-79b5-42d0-a26e-5eebdf602765, Runtime Package ID: a11c45c3-c3c1-4390-a604-bb52ce64ec93, Scope: Global, Version: 21
.0.45762.0) and in extension ('c50a4bf0-db51-4ad2-88d5-fe2287da0eb8' 'VAT Group Management', Publisher: Microsoft, Packa
ge ID: 177203f4-79b5-42d0-a26e-5eebdf602765, Runtime Package ID: a11c45c3-c3c1-4390-a604-bb52ce64ec93, Scope: Global, Ve
rsion: 21.0.45762.0).
ProcessId: 948
Tag: 0000ES3
ThreadId: 11
CounterInformation:
CustomParameters: {
}
GatewayCorrelationId:

```

**Scripts used to create container and cause the issue**
```
Write-Host "Creating" $container "from" ($navurl -replace "\?.*", "?stuff=[REDACTED]")

New-BcContainer `
          -accept_eula `
          -artifactUrl $navurl `
          -auth NavUserPassword `
          -containerName $container `
          -multitenant:$true `
          -useBestContainerOS `
		  -updateHosts `
          -Credential $creds `
          -AdditionalParameters @(
            "-v c:\temp:c:\temp", "-v c:\tvt:c:\tvt",
            "--publish 80:80",
            "--publish 443:443", 
            "--publish 7045-7049:7045-7049"
            ) `

#         -licenseFile 'C:\tvt\tvtfin.flf' `
```

**Full output of scripts**
```
PS C:\tvt> .\new-bcserver.ps1 master
Fetching artifact name
URL https://bcinsider.azureedge.net/sandbox/21.0.45762.0/gb?stuff=[REDACTED]
Set user as TVision
Creating bcserver from https://bcinsider.azureedge.net/sandbox/21.0.45762.0/gb?stuff=[REDACTED]
BcContainerHelper is version 3.0.12
BcContainerHelper is running as administrator
Hyper-V is Enabled
UsePsSession is True
Host is Microsoft Windows Server 2016 Standard - ltsc2016
Docker Client Version is 20.10.6
Docker Server Version is 20.10.6
Removing Session bcserver
Removing container bcserver
Removing entries from hosts
Removing bcserver from container hosts file
Removing bcserver-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\bcserver
Fetching all docker images
Fetching all docker volumes
Using image mcr.microsoft.com/businesscentral:10.0.14393.4104
Creating Container bcserver
Style: sandbox
Multitenant: Yes
Version: 21.0.45762.0
Platform: 21.0.45724.0
Generic Tag: 1.0.1.7
Container OS Version: 10.0.14393.4104 (ltsc2016)
Host OS Version: 10.0.14393.4104 (ltsc2016)
Using process isolation
Using locale en-GB
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Additional Parameters:
-v c:\temp:c:\temp
-v c:\tvt:c:\tvt
--publish 80:80
--publish 443:443
--publish 7045-7049:7045-7049
Files in C:\ProgramData\BcContainerHelper\Extensions\bcserver\my:
- AdditionalOutput.ps1
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container bcserver from image mcr.microsoft.com/businesscentral:10.0.14393.4104
47e6aa700bfc897042d452b52e0453bbb2c17706b6928b0c5fd46ef1185edb1f
Waiting for container bcserver to be ready
Using artifactUrl https://bcinsider.azureedge.net/sandbox/21.0.45762.0/gb
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\21.0.45762.0\platform\ServiceTier\Program Files
c:\dl\sandbox\21.0.45762.0\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\21.0.45762.0\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\21.0.45762.0\platform\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\sandbox\21.0.45762.0\gb\BusinessCentral-GB.bak
Restoring CRONUS Demo Database
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Copying Web Client Files
c:\dl\sandbox\21.0.45762.0\platform\WebClient\Microsoft Dynamics NAV
Copying ModernDev Files
c:\dl\sandbox\21.0.45762.0\platform
c:\dl\sandbox\21.0.45762.0\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\21.0.45762.0\gb\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\21.0.45762.0\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\21.0.45762.0\gb\Extensions
Copying Applications
C:\dl\sandbox\21.0.45762.0\platform\Applications
Copying Applications.GB
C:\dl\sandbox\21.0.45762.0\gb\Applications.GB
Starting Business Central Service Tier
Importing license file
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting tenant database
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Stopping Business Central Service Tier
Installation took 249 seconds
Installation complete
Initializing...
Setting host.docker.internal to -.-.-.140 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to -.-.-.140 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.27.128.1 in container hosts file
Starting Container
Hostname is bcserver
PublicDnsName is bcserver
Using NavUserPassword Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 7553027BF3C856888F258E8D414A1B37DDF90ADA
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Failed to start service 'Dynamics 365 Business Central Server [BC] (MicrosoftDynamicsNavServer$BC)'.
at <ScriptBlock>, C:\Run\navstart.ps1: line 156
at <ScriptBlock>, C:\Run\start.ps1: line 363
at <ScriptBlock>, <No file>: line 1Error
Using artifactUrl https://bcinsider.azureedge.net/sandbox/21.0.45762.0/gb
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\21.0.45762.0\platform\ServiceTier\Program Files
c:\dl\sandbox\21.0.45762.0\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\21.0.45762.0\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\21.0.45762.0\platform\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\sandbox\21.0.45762.0\gb\BusinessCentral-GB.bak
Restoring CRONUS Demo Database
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Copying Web Client Files
c:\dl\sandbox\21.0.45762.0\platform\WebClient\Microsoft Dynamics NAV
Copying ModernDev Files
c:\dl\sandbox\21.0.45762.0\platform
c:\dl\sandbox\21.0.45762.0\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\21.0.45762.0\gb\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\21.0.45762.0\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\21.0.45762.0\gb\Extensions
Copying Applications
C:\dl\sandbox\21.0.45762.0\platform\Applications
Copying Applications.GB
C:\dl\sandbox\21.0.45762.0\gb\Applications.GB
Starting Business Central Service Tier
Importing license file
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting tenant database
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Stopping Business Central Service Tier
Installation took 249 seconds
Installation complete
Initializing...
Setting host.docker.internal to -.-.-.140 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to -.-.-.140 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.27.128.1 in container hosts file
Starting Container
Hostname is bcserver
PublicDnsName is bcserver
Using NavUserPassword Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 7553027BF3C856888F258E8D414A1B37DDF90ADA
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Failed to start service 'Dynamics 365 Business Central Server [BC] (MicrosoftDynamicsNavServer$BC)'.
at <ScriptBlock>, C:\Run\navstart.ps1: line 156
at <ScriptBlock>, C:\Run\start.ps1: line 363
at <ScriptBlock>, <No file>: line 1
New-BcContainer Telemetry Correlation Id: 1a9bf86a-24ee-47d6-9527-a05a01c26fb2
Initialization of container bcserver failed
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.12\ContainerHandling\Wait-NavContainerReady.ps1:42
char:17
+ ...             throw "Initialization of container $containerName failed" ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Initialization ...bcserver failed:String) [], RuntimeException
    + FullyQualifiedErrorId : Initialization of container bcserver failed

PS C:\tvt>
...
```

**Screenshots**

**Additional context**
- does it happen all the time? --> Yes
- did it use to work? --> Yes

**Comment by freddydk** [comment]
please add -alwayspull and retry

**Comment by rdebath** [comment]
Yup looks like they _replaced_ the existing image 13 days ago.
Doing a docker-pull of the windows image allowed it to complete successfully (both single and multi-tenant).

I'll add it to my checklist, thanks.

**Comment by freddydk** [comment]
I will add to the next containerhelper that if you have an old generic (pre 1.0.2.13) - I will pull - as this is needed for 21.x

**Comment by freddydk** [comment]
Shipped in 4.0.3
